texastribune/donations
# Portal + Vue

#### What's this PR do?
+ Adds route-based data fetching capabilities
+ Adds route-based loading indicators
+ Adds error handling
+ Adds one-time-only welcome messages
+ Adds page `<title>` changes on route changes
+ Adds `robots.txt`

#### Why are we doing this? How does it help us?
More UMPing.

#### How should this be manually tested?
+ Ensure you're logged out of any Auth0 test-tenant non-staff accounts
+ Go to `/user-portal`. Confirm you get redirected to our universal log-in page.
+ Log in. Confirm you get redirected back to `/user-portal`. You should see some navigation links at the top, two welcome messages and a list of Texas Tribune headlines (a temporary call to our API to illustrate that route-based data fetching is working).
+ Click "close" above both welcome messages. They should disappear.
+ Navigate to "Blast payments." Confirm you get redirected to an error page (I temporarily have an error in that route's `created()` hook for PR review purposes).
+ Click "home" and confirm a full page refresh occurs and that you end up back on the portal home page. Confirm the welcome messages are gone.
+ Click around a few more routes and confirm the page title changes as you navigate.
+ Go to `/robots.txt` and confirm you see this: `Disallow: /user-portal/*`

#### How should this change be communicated to end users?
NA.

#### Are there any smells or added technical debt to note?
NA.

#### What are the relevant tickets?
NA.

#### Have you done the following, if applicable:
* [ ] Added automated tests? *(later)*
* [ ] Tested manually on mobile? *(later)*
* [ ] Checked for performance implications? *(later)*
* [ ] Checked for security implications? *(later)*
* [ ] Updated the documentation/wiki? *(later)*
